### DISCUSSION TITLE
What's the Mojo package format name?

### DISCUSSION THREAD

**Discussion by drunkwcodes** [modular/modular#479] [Ideas]
### Good to hear that mojo supports package structure. Now let's give a name to mojo package!

I suggest "pit" aka [fire pit](https://www.google.com/search?q=fire+pit) for the format name suffix.
Better suggestions are welcome!

The format design can borrow from [python wheel format](https://peps.python.org/pep-0427/). But that's the detail.

P.S I'm interested in constructing a Mojo Package Index website. if Modular team agrees, please notify me via gmail:
`drunkwcodes@gmail.com`. Thanks.

**Comment by cjohnson318** [comment]
Pit makes sense if there's a fire metaphor, but it also has negative connotations, e.g., these packages are the pits, pit of despair, etc. Maybe "sparks", but that might be confusing with Spark.

If the theme is "mojo", then maybe "spell" for a package name?

I think the safest choice is to just call them "packages" or "mojo packages".

**Comment by drunkwcodes** [comment]
@cjohnson318 Thanks for your reply. My English is not that good.

It should have a name.
Here are ChatGPT's suggestions:

Here are some containers with magic listed in English:

1. Magic Cauldron
2. Enchanted Vessel
3. Sorcerer's Jar
4. Enchanted Box
5. Mystic Amulet
6. Spellbound Chest
7. Enchanted Goblet
8. Warded Satchel
9. Wizard's Grimoire
10. Enchanted Bag

I found Amulet and Grimoire are interesting.

.amu and .grm, are these abbr. appropriate?

**Reply by cjohnson318** [reply]
Ha! Your English is amazing. I realize how difficult it is to think and communicate in another language.

**Comment by lattner** [comment]
📦  or 🎁 ?  :)

**Reply by UnionChips** [reply]
I like “Presents” 🎁

**Reply by drunkwcodes** [reply]
I prefer 🎁 too. But 📦 is 3D and special enough to identify as mojo package.

What about CharmedCore: A package format with a "charmed" core that ensures smooth and reliable application operations?

**Reply by helehex** [reply]
kit

**Comment by drunkwcodes** [comment]
Google Bard's replies have more kinds:

Ring (e.g., the ring that gives the wearer the power to control the weather in the story of "The Lord of the Rings")


The "Ring" is interesting, the ring Sauron is making is certainly precious.

![](https://i.playground.ru/p/4IypW0FBUnn8d-I-tal5aA.jpeg)

**Comment by drunkwcodes** [comment]
_**Mojo is akin to a mystical dialect of Python**, shrouded in enigmatic allure. Its type system and advanced features transcend the ordinary, seemingly like an esoteric language only decipherable by the enlightened. **Sounds like speaking elvish.**_

_For any daring Pythonista willing to embark on the quest of mastering Mojo, they shall wield the power to **forge the "rings" that bestow might upon the AI ecosystem, enriching lives with unparalleled wonders.**_

> Google Bard's replies have more kinds:
> 
> Ring (e.g., the ring that gives the wearer the power to control the weather in the story of "The Lord of the Rings")
> 
> The "Ring" is interesting, the ring Sauron is making is certainly precious.
> 
> ![](https://camo.githubusercontent.com/bf0625f2a1a15dcec2b7a45cdfc3a74d9ed0e4932fd771b625b9a6bf153b3d38/68747470733a2f2f692e706c617967726f756e642e72752f702f3449797057304642556e6e38642d492d74616c3561412e6a706567)

**Reply by helehex** [reply]
I want to embark on a quest after reading this

**Comment by helehex** [comment]
Kit 🎒 (🎁🧰📦🪤🍱)
Mojo kit
Kit compendium
Dictionary definition: A set of articles or equipment needed for a specific purpose.
When embarking on a quest you bring the necessary kits

**Reply by helehex** [reply]
You could categorize kits in the compendium for easy browsing:
Survival kits for journeys outside of the stdlib.
Fire kits for improving performance from pythons.
Magic kits for Artificial intelligence wizardry.
Confectionary kits for when you venture into strange territory.
Tool kits for everything else

**Comment by drunkwcodes** [comment]
The name of project metadata is also important!

`pyproject.toml` is defined in PEP 621. I'm going to clone that PEP and name it `mojoproject.toml`. Is it OK?

**Reply by s3drik** [reply]
🔥 project.toml

**Reply by drunkwcodes** [reply]
The project is on fire!

**Comment by drunkwcodes** [comment]
MojoPI proudly unveils its new logo! 
========================

The circular "M" represents the Mojo Ring, while the circuit-like design at the center, adorned with wings similar to LLVM's dragon logo, signifies the integration of MLIR logic into potentially every individual package. Each Mojo Ring achieves a level of abstraction as advanced as Python, while delving down to the intricacies of LLVM intrinsics.

So electric, so meaningful!

![mojopi_logo](https://github.com/modularml/mojo/assets/36228443/c296970b-3bff-43fe-a9c6-73c2302d23c3)
